r3bdata/califaData/R3BCalifaHitData.h

```diff
@@ -27,73 +27,74 @@ class R3BCalifaHitData : public FairMultiLinkedData
     R3BCalifaHitData();
 
     /** Constructor with arguments
-     *@param fNbOfCrystalHits		Crystal unique identifier
      *@param fEnergy				Total energy deposited
      *@param fTheta					Reconstructed theta
      *@param fPhi					Reconstructed phi
      **/
-    R3BCalifaHitData(UInt_t Nb, Double_t ene, Double_t nf, Double_t ns, Double_t theta, Double_t phi, ULong64_t time);
-
-    R3BCalifaHitData(uint64_t time, double theta, double phi, uint32_t clusterId)
-        : fTime(time)
-        , fTheta(theta)
-        , fPhi(phi)
-        , fClusterId(clusterId)
-        , fEnergy(0.)
-        , fNf(0.)
-        , fNs(0)
-        , fNbOfCrystalHits(0)
-    {
-    }
+    R3BCalifaHitData(std::vector<Int_t> crystalList, Double_t ene, Double_t nf, Double_t ns, Double_t theta, Double_t phi, ULong64_t time,Int_t clusterType);
+
+    // R3BCalifaHitData(std::vector<Int_t> crystalList, Double_t ene, Double_t nf, Double_t ns, Double_t theta, Double_t phi, ULong64_t time,Int_t clusterType)
+    //     : fCrystalList(crystalList)
+    //     , fEnergy(ene)
+    //     , fNf(nf)
+    //     , fNs(ns)
+    //     , fTheta(theta)
+    //     , fPhi(phi)
+    //     , fTime(time)
+    //     , fClusterType(clusterType)
+    //
+    // {
+    // }
 
     /** Copy constructor **/
     R3BCalifaHitData(const R3BCalifaHitData&);
 
-    /** += operator **/
-    R3BCalifaHitData& operator+=(R3BCalifaCrystalCalData& cH)
-    {
-        this->fEnergy += cH.GetEnergy();
-        this->fNf += cH.GetNf();
-        this->fNs += cH.GetNs();
-        this->fNbOfCrystalHits++;
-        return *this;
-    }
+    // /** += operator **/
+    // R3BCalifaHitData& operator+=(R3BCalifaCrystalCalData& cH)
+    // {
+    //     this->fEnergy += cH.GetEnergy();
+    //     this->fNf += cH.GetNf();
+    //     this->fNs += cH.GetNs();
+    //     this->fNbOfCrystalHits++;
+    //     return *this;
+    // }
 
     R3BCalifaHitData& operator=(const R3BCalifaHitData&) { return *this; }
 
     /** Destructor **/
     virtual ~R3BCalifaHitData() {}
 
     /** Accessors **/
-    UInt_t GetNbOfCrystalHits() const { return fNbOfCrystalHits; }
     Double_t GetEnergy() const { return fEnergy; }
     Double_t GetNf() const { return fNf; }
     Double_t GetNs() const { return fNs; }
     Double_t GetTheta() const { return fTheta; }
     Double_t GetPhi() const { return fPhi; }
     ULong64_t GetTime() const { return fTime; }
-    uint32_t GetClusterId() const { return fClusterId; }
+    uint32_t GetClusterType() const { return fClusterType; }
+    std::vector<Int_t> GetCrystalList(){return fCrystalList; }
```

Comment:
Also return the reference. Do you need to change the list later? If not, better add a const in front of this.
```suggestion
    std::vector<Int_t>& GetCrystalList(){return fCrystalList; }
```